plugins/modules/events.py

```diff
@@ -0,0 +1,135 @@
+        default: 10
+        required: false
+        type: integer
+    order:
+        description: Retrieval order for cluster events based on time of events.
+        default: ascending
+        required: false
+        type: string
+
+author:
+    - Akash Gopalakrishnan (@agopalak)
+"""
+
+EXAMPLES = r"""
+# Use argument
+- name: List cluster events
+  events
+    action: list
+    cluster_id: "deadmeat-dead-meat-dead-meatdeadmeat"
+    limit: 50
+"""
+
+RETURN = r"""
+cluster_events:
+    description: A list of cluster events
+    type: dict
+    returned: always
+    sample: [
+            {
+                "cluster_id": "46f3094d-8967-4f1d-ad09-d5fdfda3830a",
+                "event_time": "2024-11-08T20:15:44.447Z",
+                "message": "Successfully registered cluster",
+                "name": "cluster_registration_succeeded",
+                "severity": "info"
+            },
+            {
+                "cluster_id": "46f3094d-8967-4f1d-ad09-d5fdfda3830a",
+                "event_time": "2024-11-08T20:15:44.652Z",
+                "infra_env_id": "33f1638b-0419-4bc9-ac20-4baddcf14a30",
+                "message": "Updated image information (Image type is \"minimal-iso\", SSH public key is not set)",
+                "name": "image_info_updated",
+                "severity": "info"
+            },
+            {
+                "cluster_id": "46f3094d-8967-4f1d-ad09-d5fdfda3830a",
+                "event_time": "2024-11-08T20:15:52.436Z",
+                "message": "Updated status of the cluster to pending-for-input",
+                "name": "cluster_status_updated",
+                "severity": "info"
+            }
+        ]
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+API_VERSION = "v2"
+API_URL = f"https://api.openshift.com/api/assisted-install/{API_VERSION}"
+
+# add additional query parameters to the query_params_list
+QUERY_PARAMS_LIST = ["limit", "order"]
+
+def run_module():
+    module_args = dict(
+        action=dict(type="str", required=False, default="list"),
+        cluster_id=dict(type="str", required=False),
+        # any API query parameters may have to be added here
+        limit=dict(type="int", required=False, default=10),
+        order=dict(type="str",required=False, default="ascending"),
+    )
+
+    token = os.environ.get('AI_API_TOKEN')
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=False)
+
+    # Set headers
+    headers = {
+        'Content-Type': 'application/json',
+        'Authorization': f'Bearer {token}'
+    }
+
+    # List cluster events
+    if module.params.get('action') == "list":
+        list_params = {}
+        for k in QUERY_PARAMS_LIST:
+            val = module.params.get(k)
+            if val:
+                list_params = list_params | { k: val }
+
+        response = requests.get(f"{API_URL}/events?cluster_id={module.params.get('cluster_id')}", params=list_params, headers=headers)
```

Comment:
I'm curious why `cluster_id` is being specified separately in the query string rather than just passing it as another one of the `params`.

Reply:
Thanks for the catch, I have updated the code to reflect `cluster_id` as an implicit query parameter.